charts/victoria-metrics-agent/_index.md.gotmpl

```diff
@@ -50,6 +50,131 @@ remoteWrite:
 
 {{ include "chart.helmDocs" . }}
 
+## Examples
+
+### Daemonset mode
+
+ It's possible to configure vmagent to use [DaemonSet](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/) instead of Deployment and StatefulSet.
+
+Key features:
+* reduce network traffic for metric scrapping.
+* spread load for metrics collection.
+* provide resilience for single pod failure.
+
+ In this scenario, vmagent's pods will be launched on each Kubernetes Node. It requires to manually adjust scrape configuration with `spec.nodeName` pod [field selector](https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/#list-of-supported-fields) for Kubernetes API requests.
+vmagent must configuration must have either `role: pod` or `role: node`. Other `role` values will lead to excessive CPU and Memory usage, and potentially may overload Kubernetes API server.
+An example of configuration:
+```yaml
+mode: daemonSet
+
+env:
+  - name: KUBE_NODE_NAME
+    valueFrom:
+      fieldRef:
+        fieldPath: spec.nodeName
+
+remoteWrite:
+   # replace with your remote write url
+  - url: http://vmsingle-vms-victoria-metrics-k8s-stack:8428/api/v1/write
+
+config:
+  global:
+    scrape_interval: 10s
+
+  scrape_configs:
+    - job_name: vmagent
+      static_configs:
+        - targets: ["localhost:8429"]
```

Comment:
Won't this create ambigous series for each vmagent instance so it will be hard to use?
I'd suggest to add a relabeling to include some unique identifier (probaably, node name that was already passed via env) to make it easier to use.

Reply:
Good point, I think, it's better to remove it from example.